maven-embedder/src/main/java/org/apache/maven/cli/CleanArgument.java

```diff
@@ -56,30 +56,12 @@
             {
                 String cleanArgPart = arg.substring( 0, arg.length() - 1 );
 
-                // if we're building an argument, keep doing so.
-                if ( currentArg != null )
```

Comment:
How do you know that `currentArg` isn't null?

Reply:
for the previous conditions. When it reach that line always would't different of  null. 

`    if ( arg.startsWith( "\"" ) )`  the the are object is created ` currentArg = new StringBuilder( arg.substring( 1 ) );`
otherwise will never reach the next conditions 
`    if ( addedToBuffer && arg.endsWith( "\"" ) )`